bench/src/main/scala/org/http4s/bench/HeadersBench.scala

```diff
@@ -0,0 +1,35 @@
+package org.http4s
+package bench
+
+import cats.effect.IO
+import org.http4s.headers._
+import org.openjdk.jmh.annotations._
+
+import java.util.concurrent.TimeUnit
+
+// sbt "bench/jmh:run -i 5 -wi 5 -f 1 -t 1 org.http4s.bench.HeadersBench"
+@BenchmarkMode(Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Benchmark)
+class HeadersBench {
+  val baseMessage: Response[IO] = Response[IO]()
+
+  val cachedEncoding: `Content-Encoding` = `Content-Encoding`(ContentCoding.gzip)
+  val cachedServer: Server = Server(ProductId("Benchmark"))
```

Comment:
Some headers can be created once and 'cached'. This is especially relevant for these two headers, and probably for many more.

Reply:
A former committer disagreed heavily with caching values on the header companions, and I never understood, and still don't.  There is a a lot of low-hanging fruit here, probably.